commands/role.bee.inc

```diff
@@ -0,0 +1,387 @@
+      'callback' => 'role_add_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'role' => bt('Role to add'),
+      ),
+      'aliases' => array('rcrt'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-create manager' => bt("Add role 'manager'."),
+      ),
+    ),
+    'role-delete' => array(
+      'description' => bt('Delete a role.'),
+      'callback' => 'role_delete_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'role' => bt('Role to delete'),
+      ),
+      'aliases' => array('rdel'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-delete manager' => bt("Delete role 'manager'."),
+      ),
+    ),
+    'role-add-perm' => array(
+      'description' => bt('Grant specified permission(s) to a role.'),
+      'callback' => 'role_add_permission_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'permissions' => bt('Permissions'),
+        'role' => bt('Role'),
+      ),
+      'aliases' => array('rap'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-add-perm  \'post comments\' \'anonymous user\'' => bt("Allow anon users to post comments."),
+        'bee role-add-perm  "\'view own unpublished content\' , \'view any unpublished content\' , \'view revisions\'" \'anonymous user\'' => bt("Grant multiple permissions to the anon users"),
+      ),
+    ),
+    'role-remove-perm' => array(
+      'description' => bt('Remove specified permission(s) from a role.'),
+      'callback' => 'role_remove_permission_bee_callback',
+      'group' => 'roles',
+      'arguments' => array(
+        'permissions' => bt('Permissions'),
+        'role' => bt('Role'),
+      ),
+      'aliases' => array('rrp'),
+      'bootstrap' => BEE_BOOTSTRAP_FULL,
+      'examples' => array(
+        'bee role-remove-perm \'access content\' \'anonymous user\'' => bt("Hide content from anon users."),
+        'bee role-remove-perm  "\'view own unpublished content\' , \'view any unpublished content\' , \'view revisions\'" \'anonymous user\'' => bt("Remove multiple permissions from the anon users"),
+      ),
+    ),
+  );
+}
+
+/**
+ * Command callback: List all roles.
+ */
+function roles_bee_callback($arguments, $options) {
+  $rows = array();
+  $roles = user_roles(TRUE);
+  unset($roles['authenticated']);
+  if (!empty($options['role'])) {
+    // If a role has been specified, check if exists, and return error if
+    // not.
+    $role = strtolower($options['role']);
+    if (!array_key_exists($role, $roles)) {
+      bee_message(bt("The role '!role' doesn't exists!", array(
+        '!role' => $role,
+      )), 'error');
+      return;
+    }
+    $role_array = array(strtolower($options['role']));
+    $permissions = implode('\',\'', user_role_permissions($role_array));
+    if (!empty($permissions)) {
+      $permissions = '\''.$permissions.'\'';
```

Comment:
When concatenating strings, always include spaces around the joining dot and the strings/variables
```suggestion
      $permissions = '\'' . $permissions . '\'';
```